src/value/borrowed.rs

```diff
@@ -143,38 +164,35 @@ impl<'val> Struct for BorrowedStruct<'val> {
     fn iter<'a>(
         &'a self,
     ) -> Box<dyn Iterator<Item = (&'a Self::FieldName, &'a Self::Element)> + 'a> {
-        // convert (&k, &v[0..n]) -> (&k, &v[0])
-        // flattens the fields_with_text_key HashMap and chains with fields_with_no_text_key
+        // flattens the text_fields HashMap and chains with no_text_fields
         // to return all fields with iterator
         Box::new(
             self.text_fields
                 .values()
                 .flat_map(|v| v)
                 .into_iter()
                 .chain(self.no_text_fields.iter())
-                .map(|(k, v)| (k, v)),
+                .map(|(s, v)| (s, v)),
         )
     }
 
-    fn get(&self, field_name: &str) -> Option<&Self::Element> {
-        match self.text_fields.get(field_name)?.last() {
-            None => None,
-            Some(value) => Some(&value.1),
-        }
+    fn get<T: AsRef<str>>(&self, field_name: T) -> Option<&Self::Element> {
+        self.text_fields
+            .get(field_name.as_ref())?
+            .last()
+            .map(|(_s, v)| v)
```

Comment:
Made this more concise--feel free to disagree.

Reply:
No! I think this makes sense as per the parameters to the method.